File: clojure/word-count/src/word_count.clj
(ns word-count
  (:require [clojure.string :as str]))

(defn word-count
  "Given a string, return the count of each word's occurence."
  [words]
  (as-> (str/lower-case words) s
    (str/replace s #"[^a-z\s\d]" "")
    (str/split s #" ")
    (remove empty? s)
    (frequencies s)))

File: julia/word-count/word-count.jl
function wordcount(sentence)
    dict = Dict()
    words = sentence |>
        lowercase |>
        s -> replace(s, r"(?!\b'\b)[^\w ]" => " ") |>
        s -> split(s, " ") |>
        s -> filter(!isempty, s)

    for word in words
        haskey(dict, word) ? dict[word] += 1 : dict[word] = 1
    end

    return dict
end
